docs/standard/library-guidance/breaking-changes.md

```diff
@@ -0,0 +1,97 @@
+
+Adding features and improving bad behaviors is a good thing, but without care it can make it very hard for existing software to upgrade. One approach to helping developers deal with behavior breaking changes is to hide them behind settings. Settings let developers update to the latest version of your library while at the same time choosing to opt-in or opt-out of breaking changes. This strategy lets developers stay up to date while letting their consuming code adapt over time.
+
+For example, ASP.NET Core MVC has the concept of a [compatibility version](https://docs.microsoft.com/en-us/aspnet/core/mvc/compatibility-version) that modifies the features enabled and disabled on `MvcOptions`.
+
+**✔️ CONSIDER** leaving new features off by default if they affect existing users, and let developers opt-in to the feature with a setting.
+
+### Binary breaking change
+
+A binary breaking change happens when you change the public API of a library so assemblies compiled against older versions are no longer able to call it. For example, changing a method's signature by adding a new parameter will cause assemblies compiled against the older version of the library to throw a `MissingMethodException`.
+
+A binary breaking change can also break an **entire assembly**. Renaming an assembly with `AssemblyNameAttribute` will change the assembly's identity, as will adding, removing, or changing an assembly's strong naming key. A change of an assembly's identity will break all compiled code that uses it.
+
+**❌ DO NOT** change an assembly name.
+
+**❌ DO NOT** add, remove, or change the strong naming key.
+
+**✔️ CONSIDER** using abstract base classes instead of interfaces.
+
+> Adding anything to an interface will cause existing types that implement it to fail. An abstract base class allows you to add a default virtual implementation.
+
+**✔️ CONSIDER** placing the `ObsoleteAttribute` on types and members that you intend to remove. The attribute should have instructions for updating code to no longer use the obsolete API.
+
+> Code that calls types and methods with the `ObsoleteAttribute` will generate a build warning with the message supplied to the attribute. The warnings give people who use the obsolete API surface time to migrate so that when the obsolete API is removed, most are no longer be using it.
+
+```csharp
+public class Document
+{
+    [Obsolete("LoadDocument(string) is obsolete. Use LoadDocument(Uri) instead.")]
+    public static Document LoadDocument(string uri)
+    {
+        return LoadDocument(new Uri(uri));
+    }
+
+    public static Document LoadDocument(Uri uri)
+    {
+        // Load the document
+    }
+}
+```
+
+>[!div class="step-by-step"]
+[Previous](./versioning.md)
+
+## See also
```

Comment:
@JamesNK @BillWagner  Is it worth to add the link to 
https://docs.microsoft.com/en-us/dotnet/csharp/whats-new/version-update-considerations

that also discusses both source and binary compatible changes?

Reply:
Done